recipe/build.sh

```diff
@@ -13,9 +13,6 @@ if [[ "${target_platform}" == linux-* ]]; then
     # Looks like there's a bug in Opencv 3.2.0 for building with FFMPEG
     # with GCC opencv/issues/8097
     export CXXFLAGS="$CXXFLAGS -D__STDC_CONSTANT_MACROS"
-
-    export CPPFLAGS="${CPPFLAGS//-std=c++17/-std=c++11}"
-    export CXXFLAGS="${CXXFLAGS//-std=c++17/-std=c++11}"
```

Comment:
Can we please just add `-DCMAKE_CXX_STANDARD=17` to the main invocation below? It's absolutely required for abseil's ABI to work out, and even if opencv happens to do the right thing now, I'd like to be explicit about this in the recipe.

Reply:
should we bring back the `sed` commands too? I feel like opencv was overwriting the options, but maybe I don't understand cmake too well.

Reply:
Since opencv [supports](https://github.com/opencv/opencv/blob/4.x/CMakeLists.txt) being built by CMake, they really shouldn't override `CMAKE_CXX_STANDARDS`. Based on a quick search upstream, they're not touching that, so we should be fine.

I also just noticed that we (seemingly) have our own [CMakeLists.txt](https://github.com/conda-forge/opencv-feedstock/blob/main/recipe/CMakeLists.txt), but I guess that's just for an integration test. It'd be cleaner to put this into a separate folder that's hard to mistake for a build file ([this](https://github.com/conda-forge/libprotobuf-feedstock/commit/2d2a6350d7032e409d5f047aa65748478f906307) is how I did something similar for protobuf). In any case, that test needs also needs to switch to C++17.

Reply:
I add `CMAKE_CXX_STANDARD` on Windows and Linux/macOS, and the set commands just on Linux/macOS. Let me know if you think this is ok.

Reply:
I opened an issue upstream: https://github.com/opencv/opencv/issues/23752